File: haskell/src/Solutions/Euler190.hs
module Solutions.Euler190 where

maxP m' =
    let m = fromIntegral m'
    in floor $ product (map (\x -> x**x) [2..m]) * (2 / (m + 1))**(m * (m + 1) / 2)

euler190 :: IO String
euler190 = return $ show $ sum $ map maxP [2..15]

File: rust/src/solutions/euler190.rs
pub fn euler190() -> String {
    (2..16)
        .map(|_m| {
            let m = _m as f64;
            ((2..=_m).map(|m| (m as f64).powi(m)).product::<f64>()
                * (2.0 / (m + 1.0)).powf(m * (m + 1.0) / 2.0)) as i128
        })
        .sum::<i128>()
        .to_string()
}
